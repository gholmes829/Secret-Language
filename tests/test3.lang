num fib(num arg) {
    if arg <= 0 {
        return 0
    }
    else if (arg == 1) or (arg == 2) {
        return 1
    }
    return fib(arg - 1) + fib(arg - 2)
}

num main() {
    print('Fibonacci Generator:')
    for i in 16 { print(i + " - " + fib(i)) }
    return 0
}
