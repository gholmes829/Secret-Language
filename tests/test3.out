Fibonacci Generator:
0 - 0.0
1 - 1.0
2 - 1.0
3 - 2.0
4 - 3.0
5 - 5.0
6 - 8.0
7 - 13.0
8 - 21.0
9 - 34.0
10 - 55.0
11 - 89.0
12 - 144.0
13 - 233.0
14 - 377.0
15 - 610.0
